### DISCUSSION TITLE
CLS token representation for relation extraction training

### DISCUSSION THREAD

**Discussion by ketantodi** [explosion/spaCy#7438] [Help: Coding & Implementations] [answered]
I am looking to train a relation extraction component, similar to the one shared on the [ExplosionAI YouTube page](https://www.youtube.com/watch?v=8HL-Ap5_Axo). But instead of using a RoBERTa model, I am looking to use a BERT model and concatenate the [CLS] token representation to the entity representation vector, before passing it onto the classification layer. Can someone point to the resources I can refer to or the changes required in the code for the same. I have read the spacy 3.0 documentation, but was not able to find anything which could solve the above problem.

**Comment by adrianeboyd** [comment, marked as answer]
Hi, a related discussion is #7178.

**Reply by ketantodi** [reply]
Hi @adrianeboyd I looked at the thread you mentioned, also there was a similar discussion on  #6501 . Correct me if I am wrong, currently there is no in built function for getting the [CLS] token representation, the only option is to write a custom layer to get the [CLS] token representation.

**Reply by adrianeboyd** [reply]
Right, you currently need a custom layer.
